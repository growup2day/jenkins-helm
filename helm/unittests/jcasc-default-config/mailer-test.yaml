templates:
  - config-jcasc.yaml
tests:
  - it: should include mailer config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        mailer:
          smtpHost: "smtp.anznp.co.nz"
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-mailer.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-mailer.yaml]
  - it: should include blank mailer config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-mailer.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-mailer.yaml]
  - it: should not include mailer config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            unclassified:
              mailer:
                smtpHost: ""
    asserts:
      - isNull:
          path: data.[jcasc-default-config-mailer.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
