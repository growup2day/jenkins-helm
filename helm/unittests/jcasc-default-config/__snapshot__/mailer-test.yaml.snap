should include blank mailer config in default config scripts when not configured:
  1: |
    |-
      unclassified:
        # Configure System > E-mail Notification
        mailer:
          smtpHost: ""
should include mailer config in default config scripts when configured:
  1: |
    "unclassified:\n  # Configure System > E-mail Notification\n  mailer:\n    charset:
      \"UTF-8\"\n    useSsl: false\n    useTls: false\n    smtpHost: \"smtp.anznp.co.nz\"\n
      \   smtpPort: "
